e2e/test/FileUploadE2ETests.cs

```diff
@@ -54,6 +59,90 @@ public async Task FileUpload_X509_SmallFile_Http()
+
+        [TestMethod]
+        [TestCategory("LongRunning")]
+        public async Task FileUpload_SmallFile_Http_GranularSteps_x509()
+        {
+            string filename = await GetTestFileNameAsync(FileSizeSmall).ConfigureAwait(false);
+            using FileStream fileStreamSource = new FileStream(filename, FileMode.Open, FileAccess.Read);
+            Http1TransportSettings fileUploadTransportSettings = new Http1TransportSettings();
+
+            await UploadFileGranular(fileStreamSource, filename, fileUploadTransportSettings, true).ConfigureAwait(false);
+        }
+
+        [TestMethod]
+        [TestCategory("LongRunning")]
+        public async Task FileUpload_SmallFile_Http_GranularSteps_Proxy()
+        {
+            string filename = await GetTestFileNameAsync(FileSizeSmall).ConfigureAwait(false);
+            using FileStream fileStreamSource = new FileStream(filename, FileMode.Open, FileAccess.Read);
+            Http1TransportSettings fileUploadTransportSettings = new Http1TransportSettings()
+            {
+                Proxy = new WebProxy(Configuration.IoTHub.ProxyServerAddress)
+            };
+
+            await UploadFileGranular(fileStreamSource, filename, fileUploadTransportSettings).ConfigureAwait(false);
+        }
+
+        private async Task UploadFileGranular(Stream source, string filename, Http1TransportSettings fileUploadTransportSettings, bool x509auth = false)
+        {
+            await FileNotificationTestListener.InitAsync().ConfigureAwait(false);
+
+            TestDevice testDevice = await TestDevice.GetTestDeviceAsync(
+                DevicePrefix,
+                x509auth ? TestDeviceType.X509 : TestDeviceType.Sasl).ConfigureAwait(false);
+
+            DeviceClient deviceClient;
+            ClientOptions clientOptions = new ClientOptions()
+            {
+                FileUploadTransportSettings = fileUploadTransportSettings
+            };
+
+            if (x509auth)
+            {
+                X509Certificate2 cert = Configuration.IoTHub.GetCertificateWithPrivateKey();
+
+                var auth = new DeviceAuthenticationWithX509Certificate(testDevice.Id, cert);
+                deviceClient = DeviceClient.Create(testDevice.IoTHubHostName, auth, Client.TransportType.Http1, clientOptions);
+            }
+            else
+            {
+                deviceClient = DeviceClient.CreateFromConnectionString(testDevice.ConnectionString, Client.TransportType.Http1, clientOptions);
+            }
+
+            var fileUploadSasUriRequest = new FileUploadSasUriRequest()
+            {
+                BlobName = filename
+            };
+
+            var fileUploadSasUriResponse = await deviceClient.GetFileUploadSasUri(fileUploadSasUriRequest).ConfigureAwait(false);
+
+            CloudBlockBlob blob = new CloudBlockBlob(fileUploadSasUriResponse.GetBlobUri());
+            var uploadTask = blob.UploadFromStreamAsync(source);
+            await uploadTask.ConfigureAwait(false);
```

Comment:
Does it matter which storage SDK I need to use for this feature to work?

Reply:
Nope! And that's the beauty of it all

Reply:
For our e2e tests, I just used the storage SDK version that we were already using

Reply:
let's use the non-deprecated sdk for our tests then :)

Reply:
The absolute latest one doesn't work with our tests. Remember the crash I ran into when our diagnostic logging is enabled?

I use 1 version previous in the jobs test. That one works. We can use that one.

Reply:
ah yes, I remember :(

Reply:
Yeah, we can take that fix for another PR